# Day 1: Report Repair

After saving Christmas five years in a row, you've decided to take a vacation at a nice resort on a tropical island. Surely, Christmas will go on without you.

Before you leave, the Elves in accounting just need you to fix your expense report (your puzzle input); apparently, something isn't quite adding up.

Specifically, they need you to find the two entries that sum to `2020` and then multiply those two numbers together.

For example, suppose your expense report contained the following:
```
1721
979
366
299
675
145
```
In this list, the two entries that sum to `2020` are `1721` and `299`. Multiplying them together produces `1721 * 299 = 514579`, so the correct answer is `514579`.

Of course, your expense report is much larger. Find the two entries that sum to 2020; what do you get if you multiply them together?

In [44]:
input_file = open("inputs/01.txt")
puzzle_input = input_file.read()
string_numbers = puzzle_input.split()
numbers = []
for line in string_numbers:
    numbers.append(int(line))
    
print("numbers[0] + numbers [1] = ", numbers[0] + numbers[1])
input_file.close()

def x():
    for number1 in numbers:
        for number2 in numbers:
            for number3 in numbers:
                if number1 + number2 + number3 == 2020:
                    print(number1, number2, number3, number1 * number2 * number3)
                    return
x()

numbers[0] + numbers [1] =  3396
522 855 643 286977330


In [35]:
input_file.closed

True

In [36]:
from itertools import combinations
from math import prod

In [37]:
with open("inputs/01.txt") as input_file:
    puzzle_input = input_file.read()
    
numbers = [int(entry) for entry in puzzle_input.splitlines()]

In [38]:
for combination in combinations(numbers, 2):
    if sum(combination) == 2020:
        print(combination, prod(combination))

(1018, 1002) 1020036


Your puzzle answer was `1020036`.

## Part Two

The Elves in accounting are thankful for your help; one of them even offers you a starfish coin they had left over from a past vacation. They offer you a second one if you can find three numbers in your expense report that meet the same criteria.

Using the above example again, the three entries that sum to `2020` are `979`, `366`, and `675`. Multiplying them together produces the answer, `241861950`.

In your expense report, what is the product of the three entries that sum to `2020`?

Your puzzle answer was `286977330`.